manim/mobject/geometry.py

```diff
@@ -410,11 +393,80 @@ def __init__(self, point=ORIGIN, **kwargs):
 
 
 class SmallDot(Dot):
+    """
+    A dot with small radius
+    """
+
+    CONFIG = {"radius": DEFAULT_SMALL_DOT_RADIUS}
+
+
+class AnnotationDot(Dot):
+    """
+    A dot with bigger radius and bold stroke to annotate scenes.
+    """
+
     CONFIG = {
-        "radius": DEFAULT_SMALL_DOT_RADIUS,
+        "radius": DEFAULT_DOT_RADIUS * 1.3,
+        "stroke_width": 5,
+        "stroke_color": WHITE,
+        "fill_color": BLUE,
     }
 
 
+class LabeledDot(Dot):
+    """A :class:`Dot` containing a label in its center.
+
+    Parameters
+    ----------
+    label : Union[:class:`str`, :class:`~.SingleStringMathTex`, :class:`~.Text`, :class:`~.Tex`]
+        The label of the :class:`Dot`. This is rendered as :class:`~.MathTex`
+        by default (i.e., when passing a :class:`str`), but other classes
+        representing rendered strings like :class:`~.Text` or :class:`~.Tex`
+        can be passed as well.
+
+    radius : :class:`float`
+        The radius of the :class:`Dot`. If ``None`` (the default), the radius
+        is calculated based on the size of the ``label``.
+
+    Examples
+    --------
+
+    .. manim:: SeveralLabeledDots
+        :save_last_frame:
+
+        class SeveralLabeledDots(Scene):
+            def construct(self):
+                sq = Square(fill_color=RED, fill_opacity=1)
+                self.add(sq)
+                dot1 = LabeledDot(Tex("42"))
+                dot2 = LabeledDot(MathTex("a"))
+                dot3 = LabeledDot(Text("ii", color=RED))
+                dot4 = LabeledDot("3")
+                dot1.next_to(sq, UL)
+                dot2.next_to(sq, UR)
+                dot3.next_to(sq, DL)
+                dot4.next_to(sq, DR)
+                self.add(dot1, dot2, dot3, dot4)
+                self.wait(1)
+    """
+
+    def __init__(self, label, radius=None, **kwargs) -> None:
+        if isinstance(label, str):
+            from manim import MathTex
+
+            rendered_label = MathTex(label, color=BLACK)
+        else:
+            rendered_label = label
+
+        if radius is None:
+            radius = (
+                0.1 + max(rendered_label.get_width(), rendered_label.get_height()) / 2
+            )
+        Dot.__init__(self, radius=radius, **kwargs)
```

Comment:
I suggest we set the fill opacity 0 here

Reply:
I am against it for the sake of consistency: If `Dot` is a filled circle by default, then so should `LabeledDot` be. The good thing about the way this is implemented now is that you can use any keywords you are used to from `Dot` to change `LabeledDot` as well.